### PR TITLE
Fast fix of issue #1087

### PR DESCRIPTION
**Pull request by iatanasov77**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Doc update?   | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #1087
| License       | MIT

### CODE REVIEW

**Comment by c960657** [comment]
AFAICT the constants `INTL_IDNA_VARIANT_2003` and `INTL_IDNA_VARIANT_UTS46` were introduced in PHP 5.4.0 ([this commit](https://github.com/php/php-src/commit/fa4fe639447e821a0617b80609f18eb251d4bf2b)). However, `INTL_IDNA_VARIANT_UTS46` depends on the `HAVE_46_API` compile-time flag, so I guess this is the problem.

I don't know the difference between the two variants, but in most cases they will give the same result. So I suggest we just fall back to `INTL_IDNA_VARIANT_2003`, if `INTL_IDNA_VARIANT_UTS46` is not defined.

**Comment by Rotzbua** [comment]
`INTL_IDNA_VARIANT_2003` is deprecated since php7.2
https://wiki.php.net/rfc/deprecate-and-remove-intl_idna_variant_2003

**Comment by c960657** [comment]
True, but it is still available in PHP < 8. When `INTL_IDNA_VARIANT_2003 ` is removed in PHP 8, I guess it's safe to assume that `INTL_IDNA_VARIANT_UTS46` is always defined. Otherwise `idn_to_ascii()` wont work at all.

**Comment by c960657** [comment]
Just to clarify, my suggestion is this (assuming that the constant `INTL_IDNA_VARIANT_2003` still exists in PHP 8 but is just not supported by `idn_to_ascii`):

```
return idn_to_ascii($string, 0, defined('INTL_IDNA_VARIANT_UTS46') ? INTL_IDNA_VARIANT_UTS46 : INTL_IDNA_VARIANT_2003);
```

or

```
if (function_exists('idn_to_ascii')) {
    if (defined('INTL_IDNA_VARIANT_UTS46'))
        return idn_to_ascii($string, IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46);
    } else {
        return idn_to_ascii($string);
    }
}
```

**Comment by iatanasov77** [comment]
I agree with last suggestion of Christian. I test on server with defined INTL_IDNA_VARIANT_UTS46 and on another server where not defined INTL_IDNA_VARIANT_UTS46. It's work.

File: lib/classes/Swift/AddressEncoder/IdnAddressEncoder.php [hunk 1]
```diff
@@ -52,7 +52,11 @@ public function encodeString(string $address): string
     protected function idnToAscii(string $string): string
     {
         if (function_exists('idn_to_ascii')) {
-            return idn_to_ascii($string, 0, INTL_IDNA_VARIANT_UTS46);
+            if (defined('INTL_IDNA_VARIANT_UTS46')) {
+                return idn_to_ascii($string, IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46);
+            } else {
+                return idn_to_ascii($string);
```
Comment (Rotzbua) [inline lib/classes/Swift/AddressEncoder/IdnAddressEncoder.php]: To clearify maybe add a comment that IDNA2003  is used?

File: lib/classes/Swift/AddressEncoder/IdnAddressEncoder.php [hunk 2]
```diff
@@ -52,7 +52,11 @@ public function encodeString(string $address): string
...
+            }
         }
 
         if (class_exists('TrueBV\Punycode')) {
```
Comment (Rotzbua) [inline lib/classes/Swift/AddressEncoder/IdnAddressEncoder.php]: The polyfil just uses IDNA2003 also comment?

**Comment by fabpot** [comment]
Thanks for the PR. As announced some time ago, Swiftmailer is not maintained anymore.

[Symfony Mailer](https://symfony.com/doc/current/mailer.html) is the next evolution of Swiftmailer.
It provides the same features with support for modern PHP code and support for third-party providers.

**Pull request closed without merging**